docker/domserver/scripts/start.d/50-domjudge.sh

```diff
@@ -89,17 +90,65 @@ else
 	echo "TRUSTED_PROXIES=${DOCKER_GATEWAY_IP}" >> webapp/.env.local
 fi
 
-# Add trusted proxies for Nginx
 NGINX_CONFIG_FILE=/etc/nginx/snippets/domjudge-inner
 
-# Remove the previous configuration 
+# Set up BaseURL
+
+# Fix BaseURL, Such as
+# "" -> "/"
+# "domjudge" -> "/domjudge"
+if [[ -z "${WEBAPP_BASEURL}" ]] || [[ "${WEBAPP_BASEURL:0:1}" != '/' ]]; then
+	WEBAPP_BASEURL="/${WEBAPP_BASEURL}"
+	echo "Fix WEBAPP_BASEURL ${WEBAPP_BASEURL:1} -> ${WEBAPP_BASEURL}"
+fi
+
+# Fix BaseURL, Such as
+# "/" -> "/"
+# "/domjudge/" -> "/domjudge"
+if [[ "${WEBAPP_BASEURL}" != "/" ]] && [[ "${WEBAPP_BASEURL: -1}" == '/' ]]; then
+	WEBAPP_BASEURL="${WEBAPP_BASEURL%?}"
```

Comment:
Won’t this give an error if the variable is not defined?

Reply:
Won't go wrong.

If the variable is not defined.

it will meet `[[ -z "${WEBAPP_BASEURL}" ]]`

![image](https://user-images.githubusercontent.com/38343778/143769989-b6b248fe-f4b0-45c8-a36d-eaad5fb0f9a8.png)

and the `${WEBAPP_BASEURL}` will be set to `/`